pkg/database/pagination.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"math"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/pagination"
+	"github.com/jinzhu/gorm"
+)
+
+// paginate is a helper that paginates a gorm query into the given result. In
+// addition to reflecting into the provided result, it returns a pagination
+// struct.
+//
+// If page is 0, it defaults to 1. If limit is 0, it defaults to the global
+// pagination default limit.
+func paginate(query *gorm.DB, result interface{}, page, limit uint64) (*pagination.Paginator, error) {
+	if page < 1 {
+		page = 1
+	}
+	if limit < 1 {
+		limit = pagination.DefaultLimit
+	}
+
+	var total uint64
+	if err := query.
+		Count(&total).
+		Error; err != nil {
+		return nil, err
+	}
+
+	// Calculate offset.
+	offset := uint64((page - 1) * limit)
+
+	// Always select one more than the limit so we can see if there are "more"
```

Comment:
```suggestion
	// Always select one more than the limit so we can see if there are "more"
```

Are we selecting an extra? We already did a full Count()

Reply:
Ooops - this is from a previous design. Removed.